### PR TITLE
Possibly fix bug with switching rapidly between configs

### PR DESCRIPTION
**Pull request by stellar-aria**

### CODE REVIEW

**Comment by stellar-aria** [comment]
Needs to have fixes against unchecked std::optional access (`*` and `->`) before merging

File: src/deluge/dsp/delay/delay_buffer.h [hunk 1]
```diff
@@ -120,7 +120,7 @@ class DelayBuffer {
 			return;
 		}
 		// If delay buffer spinning above sample rate...
-		if (resample_config_.value().actualSpinRate >= kMaxSampleValue) {
+		if (resample_config_->actualSpinRate >= kMaxSampleValue) {
```
Comment (m-m-adams) [inline src/deluge/dsp/delay/delay_buffer.h:123]: Will this change not add a problem in that it's now entirely unchecked?

Comment (stellar-aria) [reply src/deluge/dsp/delay/delay_buffer.h:123]: No, it's checked up at the top of the function and early returns if the optional doesn't have a value.

Comment (m-m-adams) [reply src/deluge/dsp/delay/delay_buffer.h:123]: Ok perfect

**Review by m-m-adams — APPROVED** [review]

**Comment by m-m-adams** [comment]
Fix #1406

**Pull request merged**